_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,315 @@
+---
+layout: post
+title:  "Reassign partitions in Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+Scaling down Kafka Brokers and increasing the number of topics and partitions are all pretty common tasks for Kafka users. Let's take a case, for eg. There are certain brokers in a cluster and now we want to remove a broker from the cluster. We need to make sure that the broker which is going to be removed should not have any assigned partitions. Now we require a tool which can assign the partitions from the broker to be scaled down, to the brokers which will be responsible for handling these partitions. The most convenient tool for this job is the reassignment partition tool: `kafka-reassign-partitions.sh`.
+
+<!--more-->
+
+## Partition Reassignment tool
+
+Within a broker pod, the `bin/kafka-reassign-partitions.sh` tool allows you to reassign partitions to different brokers.
+
+It has three different modes:
+
+| Mode | Description |
+| :-----------: | ------------- |
+| `generate`  | Takes a set of topics and brokers and generates a reassignment JSON file which will result in the partitions of those topics being assigned to those brokers. Because this operates on whole topics, it cannot be used when you only want to reassign some partitions for some topics.|
```

Comment:
The "reassignment JSON file" concept might be worth introducing explicitly, explaining what it is before talking about how the same one is passed between the executions of the tool.